src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled.cl

```diff
@@ -820,40 +825,52 @@ inline void FUNC(fc_bf_tiled_kernel_dyn_quan)(
 
     // =====================================================================================================================================
     // Main computation loop
-    const uint iterations = MAIN_LOOP_ELEMENTS_COUNT / (TILE_IFM * SIMD);
+    const uint iterations = MAIN_LOOP_ELEMENTS_COUNT / TILE_IFM_ELEMENTS_SIZE;  // TILE_IFM_ELEMENTS_SIZE : (TILE_IFM * SIMD)
     // Each sub-group loads 2 Batch 
-    uint idx_sglid = (sglid * TILE_K) % QUANTIZE_GROUP_SIZE;       // same index for sglid 0~7 : to tile_k direction
-    uint batch_sglid = (sglid * TILE_K) / QUANTIZE_GROUP_SIZE;     // 0 to 1 : to batch direction
+    uint idx_sglid = (sglid * TILE_K) % TILE_IFM_ELEMENTS_SIZE;       // same index for sglid 0~7 : to tile_k direction
+    uint batch_sglid = (sglid * TILE_K) / TILE_IFM_ELEMENTS_SIZE;     // 0 to 1 : to batch direction
 
+    const uint scale_pitch = TILE_IN_B_PITCH / QUANTIZE_GROUP_SIZE;
+    MAKE_VECTOR_TYPE(int, TILE_B) acc_tmp[TILE_OFM] = { };
     __attribute__((opencl_unroll_hint(1)))
     for (uint ni = 0; ni < iterations; ++ni) {
         uint in_offset = input_offset + (idx_sglid + batch_sglid * TILE_IN_B_PITCH);
         uint scale_offset = input_offset / QUANTIZE_GROUP_SIZE;
         for (uint bi = 0; bi < HALF_TILE_B; ++bi) {
             // Load quantizing info from pre-quantizing kernel
             tiled_input_0[bi] = vload4(0, &quantized_input[in_offset]);
-            de_quantize_scale[bi * 2] = scale[scale_offset];
-            de_quantize_scale[bi * 2 + 1] = scale[scale_offset+ (TILE_IN_B_PITCH/QUANTIZE_GROUP_SIZE)];
-
             // Packing : Get 4(B)x4(K) integer vector (packing to 4x1 vector)
             packed_in_0[bi] = as_int(tiled_input_0[bi]);
 
             // Next batch
             in_offset += (TILE_IN_B_PITCH * 2);
-            scale_offset += (TILE_IN_B_PITCH/QUANTIZE_GROUP_SIZE * 2);
+
+             #if !NUM_LOOP_IN_DYN_QUAN_GROUP
```

Comment:
I guess `NUM_LOOP_IN_DYN_QUAN_GROUP` can be undefined. In such case, it will fail during compilation. isn't it?